manifests/install.pp

```diff
@@ -16,14 +16,20 @@
 
   $version_dir = "${data_dir}/promtail-${promtail::version}"
   $binary_path = "${version_dir}/${release_file_name}"
+  
+  if versioncmp($promtail::version, 'v1.0.0') > 0 {
```

Comment:
```suggestion
  if versioncmp($promtail::version, 'v1.0.0') >= 0 {
```

I validated this by running the following locally with `puppet apply`:

```puppet
$versions = [
  'v0.3.0',
  'v1.0.0',
  'v1.0.1',
]

$versions.each |$v| {
  if versioncmp($v, 'v1.0.0') >= 0 {
    notice("${v} >= v1.0.0, use zip")
  } else {
    notice("${v} < v1.0.0, use gz")
  }
}
```

Reply:
Also @LDaneliukas - please squash your changes into a single commit.

Reply:
I've done `>` as `v1.0.0` is using `.gz`. we can do `>=` and adjust the version to `v1.1.0` in the comparison.

I'm away now but I can update this tomorrow.

Reply:
My bad, `>` would be correct then. All that’s needed now is lint fixes and squashing